### DISCUSSION TITLE
How to get nested params on the server side? (e.g.: `[foo]/[bar].get.ts`)

### DISCUSSION THREAD

**Discussion by BracketJohn** [nuxt/nuxt#17197] [Questions] [answered]
How can one get nested params on the server side? I could not find anything for it in the docs (https://v3.nuxtjs.org/guide/features/server-routes) and so just tried out the "naive" pages-like approach of creation a folder structure like this:
```
- api/
  - [recordType]/
    - [id].get.ts  # match any non-empty sub path to `id`
    - index.ts  # match root `/`  
```

expecting that I could now do: `curl localhost:3000/api/type-1/112`, however that returns a 404. It also does not work if I remove the `index.ts`.

Looking into the underlying https://nitro.unjs.io/guide/routing.html#examples docs I was also not able to see an example with nested parameters.

So question:
- How can I achieve nested parameters using the folder / file structure?
- Bonus: How can I then still match the empty root separately (`curl localhost:3000/api/type-1/` using my above example)

**Comment by mklueh** [comment, marked as answer]
I'm facing the same "issue". Path params are very common usually when you want to access a resource directly, and I do not want to do this with query params. 

edit:

I got it working 

![image](https://user-images.githubusercontent.com/1553757/180662891-46422ee5-3d57-49e8-893c-1ea468670286.png)



call via `/api/test/123` and accessible via `event.context.params`

**Reply by BracketJohn** [reply]
ah superb!

I tried the same but for some reason it did not work. Now it does. And you can even add a `index.ts` in the folder to match just the param `test`, perfect for a `GET /test` endpoint that for exmaple lists!
